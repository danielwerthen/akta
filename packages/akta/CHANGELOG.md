# akta

## 0.1.0

### Minor Changes

- [`eef9e3f`](https://github.com/danielwerthen/akta/commit/eef9e3f381716b41723603349c3aea175c9dc51b) Thanks [@danielwerthen](https://github.com/danielwerthen)! - Testing again

* [`73cf16a`](https://github.com/danielwerthen/akta/commit/73cf16a409f85cdf954ce5a361deabb14ee492c5) Thanks [@danielwerthen](https://github.com/danielwerthen)! - Testing agin

- [`744e209`](https://github.com/danielwerthen/akta/commit/744e209e7921628849ceaac25aff560751d0bebb) Thanks [@danielwerthen](https://github.com/danielwerthen)! - Updating repository
