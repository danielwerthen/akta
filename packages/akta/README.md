# akta

[![Stable release](https://img.shields.io/npm/v/akta.svg)](https://npm.im/akta)

Akta Framework