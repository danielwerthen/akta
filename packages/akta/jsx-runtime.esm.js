export { Fragment, jsx, jsxs } from './dist/akta.esm.js';
