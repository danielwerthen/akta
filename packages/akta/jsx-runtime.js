export { Fragment, jsx, jsxs } from './dist';
