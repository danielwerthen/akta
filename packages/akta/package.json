{
  "name": "akta",
  "version": "0.2.0",
  "description": "Akta framework",
  "author": "Daniel Lichtenstein",
  "repository": {
    "type": "git",
    "url": "git+https://github.com/danielwerthen/akta.git",
    "directory": "packages/akta"
  },
  "scripts": {
    "start": "tsdx watch --tsconfig tsconfig.build.json --verbose --noClean",
    "build": "tsdx build --tsconfig tsconfig.build.json",
    "test": "tsdx test",
    "test:mem": "node --expose-gc --harmony-weak-refs ../../node_modules/.bin/tsdx test",
    "test:watch": "node --expose-gc --harmony-weak-refs ../../node_modules/.bin/tsdx test --watch",
    "test:clear": "tsdx test --clearCache",
    "lint": "tsdx lint",
    "prepublish": "npm run build"
  },
  "main": "dist/index.js",
  "module": "dist/akta.esm.js",
  "typings": "dist/index.d.ts",
  "exports": {
    ".": {
      "import": "./dist/akta.esm.js",
      "require": "./dist/index.js"
    },
    "./jsx-runtime": {
      "import": "./jsx-runtime.js",
      "require": "./jsx-runtime.cjs.js"
    }
  },
  "files": [
    "README.md",
    "dist",
    "jsx-runtime.d.ts",
    "jsx-runtime.js"
  ],
  "dependencies": {
    "akta-utils": "^0.0.1",
    "csstype": "^3.0.8",
    "rxjs": "^7.3.0"
  },
  "publishConfig": {
    "access": "public"
  },
  "jest": {
    "testEnvironment": "jsdom",
    "watchPathIgnorePatterns": [
      "dist"
    ]
  }
}
