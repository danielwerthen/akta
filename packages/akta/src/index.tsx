import { isObservable } from 'rxjs';
import { AktaNode } from './types';
import { DependencyMap } from './dependency-map';
import { attachChildren } from './mount-ops';

export * from './dependencies';
export { createDependency, createLazyDependency } from './dependency-map';

export * from './jsx-runtime';
export * from './sync-context';
export { usePrepare, usePreparer } from './dom-ops';
export * from './types';

export function mount(element: AktaNode, root: HTMLElement): () => void {
  const rest = attachChildren(root, element, new DependencyMap());
  if (isObservable(rest)) {
    const sub = rest.subscribe();
    return () => sub.unsubscribe();
  }
  return () => void 0;
}
