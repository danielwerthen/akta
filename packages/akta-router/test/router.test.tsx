import { createMemoryHistory } from 'history';
import {
  createRouter,
  Fallback,
  Route,
  routeMatchDependency,
  Router,
} from '../src';
import { mount, useDependency } from 'akta';
import { map, Subject } from 'rxjs';
describe('Test', () => {
  it('should', () => {
    const router = createRouter(createMemoryHistory());
    let counter = 0;
    router.location.subscribe(() => {
      counter++;
    });
    router.history.push('/foobar');
    router.history.push('/foobar');
    expect(counter).toBe(3);
  });
  it('should match snapshot', async () => {
    const root = document.createElement('div');
    const hist = createMemoryHistory({ initialEntries: ['/route'] });
    const momo = new Subject<string>();
    function Comp() {
      const match = useDependency(routeMatchDependency);
      return (
        <div>
          {match.pipe(
            map(match =>
              match ? `Param id: ${match.params.id}` : 'Not matched'
            )
          )}
          {momo}
          <Route path="/foobar/42">Match</Route>
        </div>
      );
    }
    mount(
      <Router history={hist}>
        <Route path="/route">Route</Route>
        <Route path="/foobar/:id">
          <Comp />
        </Route>
        <Fallback>Fallback</Fallback>
      </Router>,
      root
    );
    expect(root).toMatchSnapshot();
    hist.push('/foobar/41');
    expect(root).toMatchSnapshot();
    momo.next('Foobar2');
    expect(root).toMatchSnapshot();
    hist.push('/foobar/42');
    expect(root).toMatchSnapshot();
    hist.push('/unmatched');
    expect(root).toMatchSnapshot();
  });
});
