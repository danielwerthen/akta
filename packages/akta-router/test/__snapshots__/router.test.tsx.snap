// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test should match snapshot 1`] = `
<div>
  <div>
    Route
    
    
  </div>
</div>
`;

exports[`Test should match snapshot 2`] = `
<div>
  <div>
    Route
    
    
  </div>
</div>
`;

exports[`Test should match snapshot 3`] = `
<div>
  <div>
    
    <div>
      Param id: 41
      Foobar2
      
    </div>
    
  </div>
</div>
`;

exports[`Test should match snapshot 4`] = `
<div>
  <div>
    
    <div>
      Param id: 42
      Foobar2
      Match
    </div>
    
  </div>
</div>
`;

exports[`Test should match snapshot 5`] = `
<div>
  <div>
    
    
    Fallback
  </div>
</div>
`;
